### PR TITLE
feat: add credentials/verify endpoint to vc-api controller

### PR DESCRIPTION
**Pull request by whitneypurdum**
- add credentials/verify endpoint to vc-api controller 
- https://energyweb.atlassian.net/browse/IVA-29

### CODE REVIEW

File: apps/vc-api/src/vc-api/vc-api.controller.ts [hunk 1]
```diff
@@ -51,6 +53,21 @@ export class VcApiController {
     return await this.vcApiService.issueCredential(issueDto);
   }
 
+  /**
+   * Verify a credential. Conforms to https://w3c-ccg.github.io/vc-api/#verify-credential
+   * @returns verification results: checks, warnings, errors
+   */
+  @Post('/credentials/verify')
+  @HttpCode(200)
+  @ApiResponse({ status: 200, description: 'Verifiable Credential successfully verified' })
```
Comment (whitneypurdum) [inline apps/vc-api/src/vc-api/vc-api.controller.ts:62]: These strings come from w3c, but not sure if we need to implement them verbatim https://w3c-ccg.github.io/vc-api/#verify-credential

File: apps/vc-api/src/vc-api/credentials/dtos/verify-credential.dto.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   * https://w3c-ccg.github.io/vc-api/issuer.html#operation/issueCredential
+   */
+  vc: VerifiableCredentialDto;
+
+  /**
+   * Parameters for verifying a verifiable credential or a verifiable presentation
+   * https://w3c-ccg.github.io/vc-api/verifier.html#operation/verifyCredential
+   * https://w3c-ccg.github.io/vc-api/verifier.html#operation/verifyPresentation
+   */
+  options: VerifyOptionsDto;
```
[the remark this answers is not in the document]
Comment (whitneypurdum) [reply apps/vc-api/src/vc-api/credentials/dtos/verify-credential.dto.ts:33]: @Harasz VerifyOptionsDto and VerifiableCredentialsDto are the types here, both of which have class validators.

[the remark this answers is not in the document]
Comment (Harasz) [reply apps/vc-api/src/vc-api/credentials/dtos/verify-credential.dto.ts:33]: But it's not validated unless you add `@ValidateNested()` and `@Type(() => VerifyOptionsDto)` and from my experience it's worth to add `@IsObject()`, `@IsDefinied()` or `@IsArray()` depends on nested type

[the remark this answers is not in the document]
Comment (whitneypurdum) [reply apps/vc-api/src/vc-api/credentials/dtos/verify-credential.dto.ts:44]: good shout, just updated

**Review by Harasz — APPROVED** [review]

**Pull request merged**